packages/blade/src/components/BottomSheet/BottomSheet.web.tsx

```diff
@@ -442,11 +442,16 @@ const _BottomSheet = ({
 
   // Disable body scroll lock when the component is unmounted forcefully
   React.useEffect(() => {
-    const lockTarget = scrollRef.current!;
+    const lockTarget = scrollRef.current;
     return () => {
-      enableBodyScroll(lockTarget);
+      if (lockTarget) {
+        enableBodyScroll(lockTarget);
+      }
     };
-  }, []);
+    // when BottomSheet is mounted with isOpen={false}, then BottomSheetBody does not set scrollRef
```

Comment:
Odd, can you share a small snippet on what was the previous behaviour / if any error that we were getting?